### PR TITLE
Improve error handling in private zone discovery

### PR DESCRIPTION
**Pull request by Miciah**
Follow-up to #99.

* `pkg/dns/aws/dns.go` (`discoverZones`): Use `kerrors.NewAggregate` to combine error values from GetResourcesPages and its callback.

### CODE REVIEW

File: pkg/dns/aws/dns.go [hunk 1]
```diff
@@ -175,10 +177,7 @@ func (m *Manager) discoverZones() error {
 				},
 			},
 		}, f)
-		if innerError != nil {
-			err = innerError
-		}
-		if err != nil {
+		if err := kerrors.NewAggregate([]error{innerError, outerError}); err != nil {
```
Comment (ironcladlou) [inline pkg/dns/aws/dns.go]: Wonder why this doesn't take varargs 🤨

**Comment by Miciah** [comment]
level=error msg="\t* module.vpc.aws_lb.api_external: 1 error occurred:"
    level=error msg="\t* aws_lb.api_external: Error creating Application Load Balancer: InvalidSubnet: VPC vpc-0d60037277a84904f has no internet gateway"
    level=error msg="\tstatus code: 400, request id: 03227403-15c9-11e9-a368-69dd23774112"

/retest

**Comment by Miciah** [comment]
For the record, I checked the operator's logs from the e2e-aws-operator job, and the "failed to discover hosted zones: couldn't find private hosted zone" errors are gone.  (I wanted to check the logs on #99, but it merged before I had the chance to look at the logs from a successful test run.)

**Pull request merged**